libnavigation-router/src/main/java/com/mapbox/navigation/route/internal/hybrid/HybridRouterHandler.kt

```diff
@@ -0,0 +1,248 @@
+package com.mapbox.navigation.route.internal.hybrid
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.base.common.logger.Logger
+import com.mapbox.base.common.logger.model.Message
+import com.mapbox.base.common.logger.model.Tag
+import com.mapbox.navigation.base.route.RouteRefreshCallback
+import com.mapbox.navigation.base.route.RouteRefreshError
+import com.mapbox.navigation.base.route.Router
+
+internal sealed class RouterHandler(
+    protected val primaryRouter: Router,
+    protected val fallbackRouter: Router,
+    protected val logger: Logger
+) {
+
+    private companion object {
+        private val TAG = Tag("Mbx-HybridRouter")
+    }
+
+    protected val primaryRouterName = primaryRouter::class.simpleName
+    protected val fallbackRouterName = primaryRouter::class.simpleName
```

Comment:
`simpleName` is obusficated by proguard(and similar tools)
```suggestion
    protected val primaryRouterName = primaryRouter::class.java.canonicalName
    protected val fallbackRouterName = primaryRouter::class.java.canonicalName
```